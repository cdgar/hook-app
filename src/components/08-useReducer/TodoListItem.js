import React from "react";
import PropTypes from "prop-types";

const TodoListItem = ({ index, todo, handleToggleTodo, handleDeleteTodo }) => {
  return (
    <li className="list-group-item"  key={todo.id}>
      <p
        onClick={() => handleToggleTodo(todo.id)}
        className={`${todo.done && "complete"}`}
      >
        {index + 1}. {todo.desc}
      </p>
      <button
        onClick={() => handleDeleteTodo(todo.id)}
        className="btn btn-danger"
      >
        Delete
      </button>
    </li>
  );
};

TodoListItem.propTypes = {
  index: PropTypes.number.isRequired,
  todo: PropTypes.object.isRequired,
  handleToggleTodo: PropTypes.func.isRequired,
  handleDeleteTodo: PropTypes.func.isRequired,
};

export default TodoListItem;
