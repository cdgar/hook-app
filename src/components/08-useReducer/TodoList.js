import React from "react";
import PropTypes from "prop-types";
import TodoListItem from "./TodoListItem";

const TodoList = ({ todos, handleToggleTodo, handleDeleteTodo }) => {
  return (
    <ul className="list-group list-group-flush">
      {todos.map((todo, index) => (
        <TodoListItem
          key={todo.id}
          index={index}
          todo={todo}
          handleToggleTodo={handleToggleTodo}
          handleDeleteTodo={handleDeleteTodo}
        />
      ))}
    </ul>
  );
};

TodoList.propTypes = {
  todos: PropTypes.array.isRequired,
  handleToggleTodo: PropTypes.func.isRequired,
  handleDeleteTodo: PropTypes.func.isRequired,
};

export default TodoList;
