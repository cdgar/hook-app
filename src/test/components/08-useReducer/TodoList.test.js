import React from 'react';
import { demoTodos } from "../../fixtures/demoTodos";
import TodoListItem from "../../../components/08-useReducer/TodoListItem";
import { shallow } from 'enzyme';
import TodoList from '../../../components/08-useReducer/TodoList';

describe('TodoList.js', () => {
    
    const handleDeleteTodo = jest.fn();
    const handleToggleTodo = jest.fn();
    const wrapper = shallow(
         <TodoList 
             todos={demoTodos} 
             handleToggleTodo={handleToggleTodo} 
             handleDeleteTodo={handleDeleteTodo} 
         />
     ); 

    test('should show TodoList', () => {
        expect(wrapper).toMatchSnapshot();
    })

    test('should show TodoListItem', () => {
         expect(wrapper.find('TodoListItem').length).toBe(demoTodos.length);
         expect(wrapper.find('TodoListItem').at(0).prop('handleToggleTodo')).toEqual(expect.any(Function));
         expect(wrapper.find('TodoListItem').at(0).prop('handleDeleteTodo')).toEqual(expect.any(Function));
     })

    
})