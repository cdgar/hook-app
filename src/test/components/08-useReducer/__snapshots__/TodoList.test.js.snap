// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TodoList.js should show TodoList 1`] = `
<ul
  className="list-group list-group-flush"
>
  <TodoListItem
    handleDeleteTodo={[MockFunction]}
    handleToggleTodo={[MockFunction]}
    index={0}
    key="1"
    todo={
      Object {
        "desc": "todo1",
        "done": false,
        "id": 1,
      }
    }
  />
  <TodoListItem
    handleDeleteTodo={[MockFunction]}
    handleToggleTodo={[MockFunction]}
    index={1}
    key="2"
    todo={
      Object {
        "desc": "todo2",
        "done": false,
        "id": 2,
      }
    }
  />
</ul>
`;
